### ISSUE TITLE
qwen3.5-9B 训练一个点的定位任务时，训练val和用vllm推理时结果不一致

### ISSUE THREAD

**Issue by leeburt** [modelscope/ms-swift#8300]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

##训练时val时的结果
图片说明，红色为预测结果
<img width="684" height="390" alt="Image" src="https://github.com/user-attachments/assets/9f435796-347b-4139-9233-bd011ae85df3" />

##vllm推理时的结果
图片说明，其中的点为vllm预测结果。
<img width="915" height="426" alt="Image" src="https://github.com/user-attachments/assets/eac908ee-174e-4358-b04e-31e65fa44f8c" />

##其他说明
- 我都是将图片放缩到1008*1008尺寸，回归的绝对坐标。
- 用的qwen3-vl且用的时swift3.12版本时没有该问题。
- 使用qwen3.5-4B会好很多基本不会有这个问题

### How to Reproduce / 如何复现

qwen3.5-9B训练一个grounding任务，不采用norm1000，使用将图片缩放到1008*1008训练，训练时验证集的没有坐标偏移，但是vllm推理时有坐标偏移。同时qwen3.5-4B没有该问题。

### Additional Information / 补充信息

##环境
`Package                                  Version
---------------------------------------- ---------------------------
absl-py                                  2.4.0
accelerate                               1.13.0
addict                                   2.4.0
aiofiles                                 24.1.0
aiohappyeyeballs                         2.6.1
aiohttp                                  3.13.3
aiosignal                                1.4.0
aliyun-python-sdk-core                   2.16.0
aliyun-python-sdk-kms                    2.16.5
annotated-doc                            0.0.4
annotated-types                          0.7.0
anthropic                                0.84.0
antlr4-python3-runtime                   4.9.3
anyio                                    4.12.1
apache-tvm-ffi                           0.1.9
astor                                    0.8.1
attrdict                                 2.0.1
attrs                                    25.4.0
av                                       16.1.0
binpacking                               2.0.1
blake3                                   1.0.8
brotli                                   1.2.0
cachetools                               7.0.3
causal_conv1d                            1.6.0
cbor2                                    5.8.0
certifi                                  2026.2.25
cffi                                     2.0.0
charset-normalizer                       3.4.5
click                                    8.3.1
cloudpickle                              3.1.2
compressed-tensors                       0.13.0
contourpy                                1.3.3
cpm-kernels                              1.0.11
crcmod                                   1.7
cryptography                             46.0.5
cuda-bindings                            13.0.3
cuda-pathfinder                          1.4.0
cuda-python                              13.0.3
cupy-cuda12x                             14.0.1
cycler                                   0.12.1
dacite                                   1.9.2
datasets                                 3.6.0
deepspeed                                0.18.7
depyf                                    0.20.0
dill                                     0.3.8
diskcache                                5.6.3
distro                                   1.9.0
dnspython                                2.8.0
docstring_parser                         0.17.0
einops                                   0.8.2
email-validator                          2.3.0
eventlet                                 0.40.4
fastapi                                  0.135.1
fastapi-cli                              0.0.24
fastapi-cloud-cli                        0.14.0
fastar                                   0.8.0
ffmpy                                    1.0.0
filelock                                 3.25.0
flash_attn                               2.8.3
flash-linear-attention                   0.4.2
flashinfer-python                        0.6.4
fonttools                                4.61.1
frozenlist                               1.8.0
fsspec                                   2025.3.0
gguf                                     0.18.0
googleapis-common-protos                 1.72.0
gradio                                   5.50.0
gradio_client                            1.14.0
greenlet                                 3.3.2
groovy                                   0.1.2
grpcio                                   1.78.0
grpcio-reflection                        1.78.0
h11                                      0.16.0
hf-xet                                   1.3.2
hjson                                    3.1.0
httpcore                                 1.0.9
httptools                                0.7.1
httpx                                    0.28.1
httpx-sse                                0.4.3
huggingface_hub                          1.5.0
idna                                     3.11
ijson                                    3.5.0
importlib_metadata                       8.7.1
interegular                              0.3.3
jieba                                    0.42.1
Jinja2                                   3.1.6
jiter                                    0.13.0
jmespath                                 0.10.0
joblib                                   1.5.3
json_repair                              0.58.4
jsonschema                               4.26.0
jsonschema-specifications                2025.9.1
kiwisolver                               1.4.9
lark                                     1.2.2
liger_kernel                             0.7.0
llguidance                               1.3.0
llvmlite                                 0.44.0
lm-format-enforcer                       0.11.3
loguru                                   0.7.3
lxml                                     6.0.2
Markdown                                 3.10.2
markdown-it-py                           4.0.0
MarkupSafe                               3.0.3
matplotlib                               3.10.8
mcp                                      1.26.0
mdurl                                    0.1.2
megatron-core                            0.16.0
mistral_common                           1.9.1
ml_dtypes                                0.5.4
model-hosting-container-standards        0.1.13
modelscope                               1.34.0
mpmath                                   1.3.0
ms_swift                                 4.0.1
msgpack                                  1.1.2
msgspec                                  0.20.0
multidict                                6.7.1
multiprocess                             0.70.16
networkx                                 3.6.1
ninja                                    1.13.0
nltk                                     3.9.3
numba                                    0.61.2
numpy                                    2.2.6
nvdlfw_inspect                           0.2.2
nvidia-cublas                            13.1.0.3
nvidia-cuda-cupti                        13.0.85
nvidia-cuda-nvrtc                        13.0.88
nvidia-cuda-runtime                      13.0.96
nvidia-cudnn-cu13                        9.15.1.9
nvidia-cudnn-frontend                    1.18.0
nvidia-cufft                             12.0.0.61
nvidia-cufile                            1.15.1.6
nvidia-curand                            10.4.0.35
nvidia-cusolver                          12.0.4.66
nvidia-cusparse                          12.6.3.3
nvidia-cusparselt-cu13                   0.8.0
nvidia-cutlass-dsl                       4.4.1
nvidia-cutlass-dsl-libs-base             4.4.1
nvidia-ml-py                             13.590.48
nvidia-nccl-cu13                         2.28.9
nvidia-nvjitlink                         13.0.88
nvidia-nvshmem-cu13                      3.4.5
nvidia-nvtx                              13.0.85
omegaconf                                2.3.0
onnx                                     1.20.1
onnx-ir                                  0.2.0
onnxscript                               0.6.2
openai                                   2.26.0
openai-harmony                           0.0.8
opencv-python-headless                   4.13.0.92
opentelemetry-api                        1.40.0
opentelemetry-exporter-otlp              1.40.0
opentelemetry-exporter-otlp-proto-common 1.40.0
opentelemetry-exporter-otlp-proto-grpc   1.40.0
opentelemetry-exporter-otlp-proto-http   1.40.0
opentelemetry-proto                      1.40.0
opentelemetry-sdk                        1.40.0
opentelemetry-semantic-conventions       0.61b0
opentelemetry-semantic-conventions-ai    0.4.15
orjson                                   3.11.7
oss2                                     2.19.1
outlines_core                            0.2.11
packaging                                26.0
pandas                                   2.3.3
partial-json-parser                      0.2.1.1.post7
PasteDeploy                              3.1.0
peft                                     0.18.1
pillow                                   11.3.0
pip                                      26.0.1
prometheus_client                        0.24.1
prometheus-fastapi-instrumentator        7.1.0
propcache                                0.4.1
protobuf                                 6.33.5
psutil                                   7.2.2
py-cpuinfo                               9.0.0
pyarrow                                  23.0.1
pybase64                                 1.4.3
pycountry                                26.2.16
pycparser                                3.0
pycryptodome                             3.23.0
pydantic                                 2.12.3
pydantic_core                            2.41.4
pydantic-extra-types                     2.11.0
pydantic-settings                        2.13.1
pydub                                    0.25.1
pyeclib                                  1.7.0
Pygments                                 2.19.2
PyJWT                                    2.11.0
pyparsing                                3.3.2
python-dateutil                          2.9.0.post0
python-dotenv                            1.2.2
python-json-logger                       4.0.0
python-multipart                         0.0.22
pytz                                     2026.1.post1
PyYAML                                   6.0.3
pyzmq                                    27.1.0
quack-kernels                            0.2.10
qwen-vl-utils                            0.0.14
ray                                      2.54.0
referencing                              0.37.0
regex                                    2026.2.28
requests                                 2.32.5
rich                                     14.3.3
rich-toolkit                             0.19.7
rignore                                  0.7.6
rouge                                    1.0.1
rpds-py                                  0.30.0
ruff                                     0.15.5
safehttpx                                0.1.7
safetensors                              0.7.0
scipy                                    1.17.1
semantic-version                         2.10.0
sentencepiece                            0.2.1
sentry-sdk                               2.54.0
setproctitle                             1.3.7
setuptools                               82.0.0
shellingham                              1.5.4
simplejson                               3.20.2
six                                      1.17.0
sniffio                                  1.3.1
sortedcontainers                         2.4.0
sse-starlette                            3.3.2
starlette                                0.52.1
supervisor                               4.3.0
swift                                    2.37.0
sympy                                    1.14.0
tabulate                                 0.10.0
tensorboard                              2.20.0
tensorboard-data-server                  0.7.2
tiktoken                                 0.12.0
tokenizers                               0.22.2
tomlkit                                  0.13.3
torch                                    2.10.0+cu130
torch_c_dlpack_ext                       0.1.5
torchaudio                               2.10.0+cu130
torchvision                              0.25.0+cu130
tqdm                                     4.67.3
transformer_engine                       2.12.0
transformer_engine_cu12                  2.12.0
transformer_engine_torch                 2.12.0
transformers                             5.2.0
transformers-stream-generator            0.0.5
triton                                   3.6.0
trl                                      0.28.0
typer                                    0.24.1
typer-slim                               0.24.0
typing_extensions                        4.15.0
typing-inspection                        0.4.2
tzdata                                   2025.3
urllib3                                  2.6.3
uv                                       0.10.8
uvicorn                                  0.41.0
uvloop                                   0.22.1
vllm                                     0.17.0rc1.dev103+g57c84ff12
watchfiles                               1.1.1
websockets                               15.0.1
Werkzeug                                 3.1.6
wheel                                    0.46.3
xattr                                    1.3.0
xgrammar                                 0.1.29
xxhash                                   3.6.0
yarl                                     1.23.0
zipp                                     3.23.0
zstandard                                0.25.0`

**Comment by zkailinzhang** [comment]
你好，我基于vqa 微调qwen3.5b-32b 基于ms-swift 报错，楼主 可以提供下微调脚本吗

**Comment by leeburt** [comment]
> 你好，我基于vqa 微调qwen3.5b-32b 基于ms-swift 报错，楼主 可以提供下微调脚本吗
MODEL_NAME="Qwen/Qwen3.5-4B"
TRAIN_CMD=(swift sft
    --model "${MODEL_NAME}"
    --tuner_type full 
    --dataset "${DATASETS[@]}"
    --load_from_cache_file true
    --add_non_thinking_prefix true
    --loss_scale ignore_empty_think
    --torch_dtype bfloat16
    --num_train_epochs 5
    --eval_metric port_prf
    --predict_with_generate true
    --per_device_train_batch_size 4
    --per_device_eval_batch_size 4
    --learning_rate 1e-5
    --gradient_accumulation_steps 1
    --group_by_length true
    --output_dir "${OUTPUT_DIR}"
    --logging_dir "${LOGGING_DIR}" \
    --eval_steps 500
    --save_steps 500
    --save_total_limit 2
    --logging_steps 10
    --max_length 7120
    --max_new_tokens 4096
    --warmup_ratio 0.05
    --dataset_num_proc 8
    --dataloader_num_workers 8
    --deepspeed zero3
    --model_author libo
    --attn_impl flash_attention_2 
    --truncation_strategy delete
    --add_version False
)

**Comment by lixumin-zai** [comment]
请问排查出问题了，谁来救救我们

**Comment by Jintao-Huang** [comment]
啊 是什么问题啊
提一个 单独的issue吧
